### PR TITLE
Ejbca refactor

### PR DESCRIPTION
**Pull request by muhamedcpqd**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
features, bug fix, tests

* **What is the current behavior?** (You can also link to an open issue here)
bugs with cache, old api, old tests

* **What is the new behavior (if this is a feature change)?**
correcting cache bug, compliance api with dojot, new/modified tests

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#1287

* **Other information**:

### CODE REVIEW

File: ejbca/src/app.js [hunk 1]
```diff
@@ -13,38 +15,45 @@ let server = {
...
 
-function initApp(clientEJBCA) {
+function initApp(clientEJBCA, messenger, dojotConfig) {
 
     server.app = express();
     server.app.use(bodyParser.json({ type: "*/*" }));
     server.app.use(bodyParser.urlencoded({ extended: true }));
 
     /* Starting the server */
     server.httpServer = server.app.listen(5583, () => {
```
Comment (mprevide) [inline ejbca/src/app.js]: Acredito que tenha q ser uma variavel de ambiente a porta

File: ejbca/src/config.js [hunk 2]
```diff
@@ -5,6 +5,65 @@ const soap = {
...
+        },
+        subjects: {
+            deviceData: process.env.DOJOT_SUBJECT_DEVICE_DATA || "device-data",
+            devices: process.env.DOJOT_SUBJECT_DEVICES || "dojot.device-manager.device",
+            tenancy: process.env.DOJOT_SUBJECT_TENANCY || "dojot.tenancy",
+        },
+    },
+    kafka: {
+        consumer: {
+            "group.id": "vernemq-group",
```
Comment (rascaraficci) [inline ejbca/src/config.js]: why this name? It seems a little weird for this service.

File: ejbca/src/config.js [hunk 3]
```diff
@@ -5,6 +5,65 @@ const soap = {
...
+            "metadata.broker.list": process.env.KAFKA_HOSTS || "kafka-server:9092",
+        },
+        dojot: {
+            connectionRetries: 10,
+            subjects: {
+                verne: "vernemq-epic-channel",
```
Comment (rascaraficci) [inline ejbca/src/config.js]: why this name? It seems a little weird for this service.

File: ejbca/src/config.js [hunk 4]
```diff
@@ -5,6 +5,65 @@ const soap = {
...
+        producer: {
+            "dr_cb": true,
+            "metadata.broker.list": process.env.KAFKA_HOSTS || "kafka-server:9092",
+            "socket.keepalive.enable": true,
+        },
+    },
+};
+
+
+const dojot = {
```
Comment (rascaraficci) [inline ejbca/src/config.js]: What is this?

File: ejbca/src/config.js (hunk 2 above)
Comment (muhamedcpqd) [reply ejbca/src/config.js]: yes, i forgot to change here..

File: ejbca/src/config.js (hunk 4 above)
Comment (muhamedcpqd) [reply ejbca/src/config.js]: just a const to set the ejbca port for the express server.

Comment (muhamedcpqd) [reply ejbca/src/config.js]: for the wsdl service.

File: ejbca/src/config.js (hunk 3 above)
Comment (muhamedcpqd) [reply ejbca/src/config.js]: this config is not necessary, i will remove it.

File: ejbca/tests/unit-tests/app.test.js [hunk 5]
```diff
@@ -30,9 +30,10 @@ describe("Testing app client", () => {
...
         });
 
-        it("Should httpServer fail and be stopped", () => {
-            if (!server.initApp(null)) { server.stopApp(); }
-            expect(server.server.isInitialized).toEqual(false);
+        it("Should initApp fail", () => {
+            server.initApp(null).catch(() => {
+                expect(server.server.isInitialized).toEqual(true);
+            })
         })
```
Comment (eduardogmisiuk) [inline ejbca/tests/unit-tests/app.test.js]: Is this test intended to catch an initialization error or not? The expect is comparing `server.server.isInitialized` to `true`

File: ejbca/package.json [hunk 6]
```diff
@@ -17,6 +17,8 @@
   "author": "Muhamed Avila",
   "license": "GPL-3.0",
   "dependencies": {
+    "@dojot/dojot-module": "0.0.1-beta.7",
```
Comment (mprevide) [inline ejbca/package.json]: Nao seria melhor testar com a versão do jonaphael ?

File: ejbca/README.md [hunk 7]
```diff
@@ -1,12 +1,44 @@
...
+
+The EJBCA persists all the data created in dojot (clients certificate, CA's p12, etc..) in  database. If the system administrator needs to export the CA information (keys and certificates) he can extract the p12 file from EJBCA with the respectively .p12 password.
+
+## EJBCA Profiles configuration
+
+Profiles of CA/clients can be configured [here](./profiles). With those profiles, the system administrator can configure parameters like certificate expiration and data-base write/read control.
+
+## API Use
+
+The Registration Authority (RA) is automatically created in EJBCA with the respectivelly .p12 file (check [here](config_ra.sh) for more configuration information). The RA communicates to EJBCA backend using SOAP. The client of dojot can communicate to EJBCA Nodejs service using the REST API. For more information about the endpoints check [here](https://dojot.github.io/ejbca-rest/apiary_latest.html).
```
Comment (mprevide) [inline ejbca/README.md]: Você ira arrumar depois https://dojot.github.io/ejbca-rest/apiary_latest.html?

File: ejbca/README.md [hunk 8]
```diff
@@ -1,12 +1,44 @@
...
+
+# **Configuration**
+
+## **Environment Variables**
+
+Before running the kubernetes environment with the mqtt bridge service, make sure you configure the environment variables to match your needs. The variables can be configured in the .yaml file of the service.
```
Comment (mprevide) [inline ejbca/README.md]: Não necessariamente kubernetes.

Comment (mprevide) [reply ejbca/README.md]: E também não deve ser especifica para kubernetes.

File: ejbca/routes/ejbcaRoute.js [hunk 9]
```diff
@@ -1,337 +1,474 @@
...
+          moreinfo: 'https://dojot.github.io/ejbca-rest/apiary_latest.html',
+        });
+      }
 
-    app.post('/user', ejbcaUtils.validators.userValidator, (req, res) => {
+      if (!cert) {
+        return res.status(404).json({
+          code: 404,
+          message: 'No certificate found',
+          moreinfo: 'https://dojot.github.io/ejbca-rest/apiary_latest.html',
```
Comment (mprevide) [inline ejbca/routes/ejbcaRoute.js]: Moreinfo, deve ser uma variavel de ambiente

File: ejbca/tests/unit-tests/app.test.js [hunk 10]
```diff
@@ -1,51 +1,47 @@
-"use strict";
-
-let fakeSoap = require('../../lib/dojot_soap');
-let server = require('../../src/app');
-const express = require("express");
+/* eslint-disable no-undef */
```
Comment (mprevide) [inline ejbca/tests/unit-tests/app.test.js]: Isso é mesmo necessário *eslint-disable no-undef*?

File: ejbca/README.md (hunk 7 above)
Comment (muhamedcpqd) [reply ejbca/README.md]: provavelmente.

**Pull request merged**